src/debot/term_browser.rs

```diff
@@ -12,103 +12,177 @@
+        };
+        let abi = load_abi(&abi_json)?;
+        {
+            let msgs = &mut callbacks_ref.state.write().unwrap().msg_queue;
+            self.msg_queue.append(msgs);
         }
 
-        loop {
-            let res = action_input(self.active_actions.len());
-            if res.is_err() {
-                println!("{}", res.unwrap_err());
-                continue;
+        self.bots.insert(
+            debot_addr,
+            DebotEntry {
+                abi,
+                dengine,
+                callbacks: callbacks_ref,
             }
-            let (n, _, _) = res.unwrap();
-            let act = self.active_actions.get(n - 1);
-            if act.is_none() {
-                println!("Invalid action. Try again.");
-                continue;
-            }
-            return act.map(|a| a.clone());
-        }
+        );
+        Ok(())
     }
 
-    async fn handle_interface_call(
-        client: TonClient,
+    async fn call_interface(
+        &mut self,
         msg: String,
-        debot: &mut DEngine,
-        interfaces: &SupportedInterfaces,
+        interface_id: &String,
+        debot_addr: &str,
     ) -> Result<(), String> {
-
-        let parsed = parse_message(
-            client.clone(),
-            ParamsOfParse { 
-                boc: msg.clone(),
-                ..Default::default()
-            },
-        )
-        .await
-        .map_err(|e| format!("{}", e))?;
-
-        let iface_addr = parsed.parsed["dst"]
-            .as_str()
-            .ok_or(format!("parsed message has no dst address"))?;
-        let wc_and_addr: Vec<_> = iface_addr.split(':').collect();
-        let interface_id = wc_and_addr[1].to_string();
-
-        if let Some(result) = interfaces.try_execute(&msg, &interface_id).await {
+        let debot = self.bots.get_mut(debot_addr)
+            .ok_or_else(|| "Internal browser error: debot not found".to_owned())?;
+        if let Some(result) = self.interfaces.try_execute(&msg, interface_id).await {
             let (func_id, return_args) = result?;
             debug!("response: {} ({})", func_id, return_args);
-            let result = debot.send(
-                iface_addr.to_owned(), func_id, return_args.to_string()
-            ).await;
+            let call_set = match func_id {
+                0 => None,
+                _ => CallSet::some_with_function_and_input(&format!("0x{:x}", func_id), return_args),
+            };
+            let response_msg = encode_internal_message(
+                self.client.clone(),
+                ParamsOfEncodeInternalMessage {
+                    abi: Some(debot.abi.clone()),
+                    address: Some(debot_addr.to_owned()),
+                    src_address: None,
+                    deploy_set: None,
+                    call_set,
+                    value: "1000000000000000".to_owned(),
+                    bounce: None,
+                    enable_ihr: None,
+                }
```

Comment:
Pls as in https://github.com/tonlabs/tonos-cli/pull/112
remove `None` and add `..Default::default()`

Reply:
OK